src/functions.php

```diff
@@ -231,6 +232,19 @@ function ($reason, $idx, Promise $aggregate) {
         ksort($results);
         return $results;
     });
+
+    if (true === $recursive) {
+        $promise = $promise->then(function ($results) use ($recursive, &$promises) {
+            foreach ($promises AS $promise) {
+                if (\GuzzleHttp\Promise\PromiseInterface::PENDING === $promise->getState()) {
+                    return all($promises, $recursive);
```

Comment:
Thanks for looking at this! Sorry, I should have clarified -- I meant these callbacks: https://github.com/guzzle/promises/blob/master/src/EachPromise.php#L33.